### PR TITLE
CryptoPkg: Additional CodeQL fixes

### PR DESCRIPTION
**Pull request by Flickdm**
## Description

Fixes for codeql package, after the filter updates these changes weren't necessary but adds null checks to unit test

- [ ] Impacts functionality?
  - No
- [ ] Impacts security?
  - No
- [ ] Breaking change?
  - No
- [x] Includes tests?
  - Yes
- [ ] Includes documentation?
  - No

## How This Was Tested

Executed Tests

## Integration Instructions

N/A

### CODE REVIEW

File: CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c [hunk 1]
```diff
@@ -324,15 +337,28 @@ TestVerifyRsaCertPkcs1SignVerify (
...
+  TestStatus = UNIT_TEST_PASSED;
+Exit:
   //
   // Release Resources.
   //
-  RsaFree (RsaPubKey);
-  RsaFree (RsaPrivKey);
-  FreePool (Signature);
-  FreePool (Subject);
+  if (RsaPrivKey != NULL) {
```
Comment (kuqin12) [inline CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c]: Set these to NULL by default?

**Review by kuqin12 — APPROVED** [review]

File: CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c [hunk 2]
```diff
@@ -324,15 +337,28 @@ TestVerifyRsaCertPkcs1SignVerify (
...
-  return UNIT_TEST_PASSED;
+  if (RsaPubKey != NULL) {
+    RsaFree (RsaPubKey);
+  }
+
+  if (Signature != NULL) {
+    FreePool (Signature);
+  }
+
+  if (Subject != NULL) {
```
Comment (makubacki) [inline CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c]: As a general practice, I suggest freeing these in reverse order of the allocations. It doesn't seem to be an issue here, but doing so can help more naturally release buffers that may be dependent on each other.

For example, (1) a structure buffer is allocated with (2) a pointer to another buffer that is allocated. Free (2), then free (1).

**Review by os-d — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

File: CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c (hunk 2 above)
Comment (Flickdm) [reply CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c]: Yeah that's totally fair

**Comment by makubacki** [comment]
@Flickdm can you update this PR and complete it?

**Comment by makubacki** [comment]
> @Flickdm can you update this PR and complete it?

I put a PR here into the branch on your fork. You should just need to complete it and this PR will be good to go.

https://github.com/Flickdm/mu_basecore/pull/1/

File: CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c (hunk 1 above)
Comment (makubacki) [reply CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c]: Completed in PR commit https://github.com/Flickdm/mu_basecore/pull/1/commits/59d446c9d3451f0c212cbb994684129a40e3e8ac

File: CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c (hunk 2 above)
Comment (makubacki) [reply CryptoPkg/Test/UnitTest/Library/BaseCryptLib/RsaPkcs7Tests.c]: Done in PR commit https://github.com/Flickdm/mu_basecore/pull/1/commits/59d446c9d3451f0c212cbb994684129a40e3e8ac

**Pull request closed without merging**